### ISSUE TITLE
Brainstorm and Create a way to Automate Updates for DRS and Record Holder Leaderboards

### ISSUE THREAD

**Issue by tehchives** [WhyDRS/wix-site-public#8]
Currently, the DRS and Record Holder leaderboards use images that I created in a photo editor, but there is likely a way we can keep the background but superimpose plaintext drawn from the Database or other repo. This would make it much easier to update (either manually or, ideally, automatically as new filings and data are released).

**Comment by jergy2** [comment]
It would be relatively simple to make the same or similar display that is currently an image file using just HTML and some CSS.

Considering that as of now this information is apparently only updated manually, you could just manually modify the text of the HTML each time an update needs to be made.  This would be easy to implement and maintain and suitable in the case of infrequent updates.

When there becomes a more active set of data that changes over time that would reflect the leaderboard info and rankings, then it would be suitable to automate this using javascript for example to fetch the information and update the text in the HTML.

**Comment by tehchives** [comment]
I communicated with jergy2 additionally on Discord and he provided the following code:

---

`<div
    style="height:100%;width: 100%; background-color: white;display:flex;flex-direction: row; justify-content: center;">
    <div class="list-container">

        <div class="numbers">
            <div style="margin-top: 30px;"></div>
            <div class="number-row">1</div>
            <div class="number-row">2</div>
            <div class="number-row">3</div>
        </div>

        <div class="list">
            <div class="title">
                <div>Ticker</div>
                <div>Record Holders</div>
            </div>
            <div class="list-item first-item">
                <div>BHF</div>
                <div>1,500,000</div>
            </div>
            <div class="list-item second-item">
                <div>PRU</div>
                <div>1,096,339</div>
            </div>
            <div class="list-item third-item">
                <div>T</div>
                <div>781,511</div>
            </div>
            <div class="list-item">
                <div>VZ</div>
                <div>462,587</div>
            </div>
            <div class="list-item">
                <div>IBM</div>
                <div>388,977</div>
            </div>
            <div class="list-item">
                <div>XOM</div>
                <div>308,630</div>
            </div>
            <div class="list-item">
                <div>CINF</div>
                <div>258,700</div>
            </div>
            <div class="list-item">
                <div>WFC</div>
                <div>231,886</div>
            </div>
            <div class="list-item">
                <div>PFG</div>
                <div>219,911</div>
            </div>
            <div class="list-item">
                <div>WMT</div>
                <div>209,870</div>
            </div>
        </div>
    </div>
`
---

which he said should output:

![Image](https://github.com/user-attachments/assets/9e9a142a-a4a6-4248-a4bc-ca9bf06f54a9)

This could be repurposed not only to update more simply, but also to solve both leaderboard update problems (and future ones as well!)

I will be poking around in the Wix devtools to try and get this injected code to function properly. Thank you so much to @jergy2 !!

**Comment by tehchives** [comment]
Okay - I've figured it out! Got a custom element working with jergy's above code. There's cleanup work to do on both pages that I don't have time to do tonight, but this is very close to ready for a data update. 

![Image](https://github.com/user-attachments/assets/571000db-8d81-4e20-9590-6a24666f1694)

**Comment by JamesAlfonse** [comment]
WOOO! Exciting stuff. Thanks @jergy2 and @tehchives !

One step closer to automating that entire process 😁

**Comment by tehchives** [comment]
I've created some files to track sourcing and host this code over in [wix-site-public](https://github.com/WhyDRS/wix-site-public/tree/main/leaderboards) - Tomorrow will update the pages themselves.

**Comment by tehchives** [comment]
I ended up not getting to it when I wanted to! Still need to pull the rest of the numbers on recent 10K, going to be working on that next to add to that repo.

**Comment by tehchives** [comment]
All set - Data for both pages is updated to the most recent available, and new code is live so that it will be significantly easier to update in the future.
@jergy2 Thank you again. When you have a chance to look at the live site as implemented I'd appreciate. To touch up, I think the side numbering looks a bit off and I tried to edit the code to fix it (just trial and error) but didn't have any luck. Any other concepts on how to improve this resource are definitely welcomed!

**Comment by jergy2** [comment]
Yes i see an issue with the alignment of the numbers and the boxes.  the original code i provided was not very good.  here is an updated copy, give this a try, it should fix the problem:

`<div class="list-container">
              <div class="list">
                  <div class="list-item ">
                      <div class="list-item-number">
                          
                      </div>
                      <div class="title">
                          <div>Ticker</div>
                          <div>Record Holders</div>
                      </div>
                  </div>
  
                  <div class="list-item ">
                      <div class="list-item-number">
                          1
                      </div>
                      <div class="list-item-box first-item">
                          <div>BHF</div>
                          <div>1,500,000</div>
                      </div>
                  </div>
                  <div class="list-item ">
                      <div class="list-item-number">
                          2
                      </div>
                      <div class="list-item-box second-item">
                          <div>PRU</div>
                          <div>1,096,339</div>
                      </div>
  
                  </div>
                  <div class="list-item ">
                      <div class="list-item-number">
                          3
                      </div>
                      <div class="list-item-box third-item">
                          <div>T</div>
                          <div>781,511</div>
                      </div>
                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box">
                          <div>VZ</div>
                          <div>462,587</div>
                      </div>
                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box">
                          <div>IBM</div>
                          <div>388,977</div>
                      </div>

                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box ">
                          <div>XOM</div>
                          <div>308,630</div>
                      </div>

                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box ">
                          <div>CINF</div>
                          <div>258,700</div>
                      </div>

                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box ">
                          <div>WFC</div>
                          <div>231,886</div>
                      </div>

                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box ">
                          <div>PFG</div>
                          <div>219,911</div>
                      </div>

                  </div>
                  <div class="list-item">
                      <div class="list-item-number">
                      </div>
                      <div class="list-item-box ">
                          <div>WMT</div>
                          <div>209,870</div>
                      </div>

                  </div>
              </div>
          </div>


            <style>

                .list-container{
                    width: 100%;
                    display:flex;
                    flex-direction: row;
                    justify-content: center;
                }
                .list {
                    display: flex;
                    flex-direction: column;
                    margin-top: 10px;                    
                    min-width: 300px;    /* adjust as needed  */
                    max-width: 500px;    /* adjust as needed  */
                }


                .list-item {
                    display: flex;
                    flex-direction: row;
                    align-items: center;
                }

                .list-item-number {
                    width: 15px;
                    padding: 5px;
                    display:flex;flex-direction: row;
                    align-items: center;
                    margin-right:20px;
                }

                .title{
                    display:flex;
                    flex-direction: row;
                    justify-content: space-between;
                    align-items: center;
                    width: 100%;
                }

                .list-item-box {

                    width: 100%;
                    display: flex;
                    flex-direction: row;
                    justify-content: space-between;
                    align-items: center;

                    background-color: rgb(20, 79, 155);
                    border-radius: 2px;
                    padding: 2px;
                    padding-left: 4px;
                    padding-right: 4px;
                    margin-bottom: 10px;
                    border: 1px solid gray;
                    font-size: 1.2em;
                    color: white;
                }

                .first-item {
                    background-color: rgb(236, 190, 8);
                    color: black;
                }

                .second-item {
                    background-color: rgb(135, 165, 201);
                    color: black;
                }

                .third-item {
                    background-color: rgb(214, 153, 64);
                    color: black;
                }
            </style>

`

**Comment by tehchives** [comment]
Is it supposed to be split like that? Also - If you would not mind, you could set a PR to provide this updated code in the https://github.com/WhyDRS/wix-site-public/blob/main/leaderboards/leaderboard%20image%20code area. That would make it easier to find and maintain for the future. Thank you for your quick help on this!

**Comment by tehchives** [comment]
Updated the leaderboards pages with the new code!

Eventually, I think we could think about ways to automate this process and expand to other datasets - but for now, we're looking great. I will close this issue.
